### DISCUSSION TITLE
[Object Diagram] How to center a multi-line title (hyperlink) of an object

### DISCUSSION THREAD

**Discussion by claudio-salvio** [plantuml/plantuml#1503] [Q&A]
Hello, 👋

In an object diagram, does anyone know of **a way to center all the lines of a multi-line title** (which is a **hyperlink**) on an object ?

Two lines hyperlink: centered **FAIL** ❌
```plantuml
@startuml
object "[[http://www.awebsite.com/FirstNameLastName FirstName\nLN]]" as FirstNameNewLineLNLink {
    <color:red>Link</color>    
}
footer <size:10>Date Created: %date( "yyyy.MM.dd' 'HH:mm" ) \nPlantUml Version: %version()
@enduml
```
[![Two lines hyperlink: centered **FAIL** ❌ at plantuml.com/plantuml](https://github.com/plantuml/plantuml/assets/66575834/d50c507d-9684-4785-9682-cbc79e3b75ff)](https://www.plantuml.com/plantuml/uml/SyfFKj2rKt3CoKnELR1Io4ZDoSa70000)

Two lines of text: centered **OK** ✔️
```plantuml
@startuml
object "FirstName\nLN" as FirstNameLN {
    <color:green>Text Only</color>    
}
footer <size:10>Date Created: %date( "yyyy.MM.dd' 'HH:mm" ) \nPlantUml Version: %version()
@enduml
```
[![Two lines of text: centered **OK** ✔️ at plantuml.com/plantuml](https://github.com/plantuml/plantuml/assets/66575834/163f897c-9705-48cd-9fa4-7f92cdfd3985)](https://www.plantuml.com/plantuml/uml/SoWkIImgAStDuSh8J4bLICuiIiv9vKhEIImkLWZEo2yeoCnBTqfCvt98pKi1gW80)

One line hyperlink and one line text: centered **OK** ✔️
```plantuml
@startuml
object "[[http://www.awebsite.com/FirstNameLastName FirstName]] \nLN" as FirstNameLinkNewLine {
    <color:green>Mixed</color>    
}
footer <size:10>Date Created: %date( "yyyy.MM.dd' 'HH:mm" ) \nPlantUml Version: %version()
@enduml
```
[![image](https://github.com/plantuml/plantuml/assets/66575834/88b025a1-f0d4-4117-8c3c-b9153f5e4639)](https://www.plantuml.com/plantuml/uml/SoWkIImgAStDuSh8J4bLICuiIiv9vKhEIImkLWZEo2yeoCnBTqfCvt98pKi1gW80)

Kind regards,
Claudio Salvio

**Comment by DRKV333** [comment]
You can split the link into multiple links, with a new line between them:
`"[[http://www.awebsite.com/FirstNameLastName FirstName]]\n[[http://www.awebsite.com/FirstNameLastName LN]]"`

Or more complicatedly, with macros:

```puml
@startuml
!function $replace($in, $what, $with)
  !$whatlen = %strlen($what)
  !$pos = %strpos($in, $what)
  !while $pos != -1
    !$in = %substr($in, 0, $pos) + $with + %substr($in, $pos + $whatlen, %strlen($in) - $pos - $whatlen)
    !$pos = %strpos($in, $what)
  !endwhile
  !return $in
!endfunction

!function $multiLink($link, $text)
  !$res = ""
  !$nl = %false()
  !$text = $replace($text, "\n", %newline())
  !foreach $line in %splitstr($text, %newline())
    !if ($nl)
      !$res = $res + "\n"
    !else
      !$nl = %true()
    !endif
    !$res = $res + "[[" + $link + " " + $line + "]]"
  !endfor
  !return $res
!endfunction

object "$multiLink("google.com", "this is a\nmultiline link")" as FirstNameNewLineLNLink {
    <color:red>Link</color>    
}
footer <size:10>Date Created: %date( "yyyy.MM.dd' 'HH:mm" ) \nPlantUml Version: %version()
@enduml
```
![XL1DQzj04BthLumNXqZ4TfEhSKoWfUIGc5xQIvp3HXf5swvspUwexWU5GlzfVqbdPvt8X4ACy7oycNbltrKa7MZeRJ5f1rUJyGwc0NTMrrXEZPl1TDzfIi5GLnK04sbOT7099v42PwMqChZpyG1mbY-CwBup5a5c9fSml-1UsZ7vrl30MtdbV2PJ5PnbOex7k822PIcpKObn5SmpFd_6](https://github.com/plantuml/plantuml/assets/30904009/6f5cb8f7-aac6-450c-85ea-87cf4b440d3b)
